File: RiptideCore/Sources/RiptideCore/ProgramPrinter.swift
public enum ProgramPrinter {
    /// Human-readable week, used by snapshot tests and future debug screens.
    public static func table(_ program: GeneratedProgram) -> String {
        var out: [String] = []
        for (i, day) in program.days.enumerated() {
            let total = day.lifts.reduce(0) { $0 + $1.sets }
            out.append("Day \(i + 1) — \(day.lifts.count) lifts, \(total) sets")
            for lift in day.lifts {
                out.append("  \(lift.exercise.name) [\(lift.exercise.primary.rawValue)] \(lift.sets) x \(lift.exercise.repRange)")
            }
        }
        for s in program.shortfalls {
            out.append("SHORTFALL \(s.muscle.rawValue): \(s.achieved) of \(s.target) target sets")
        }
        return out.joined(separator: "\n") + "\n"
    }
}

File: web/src/core/programPrinter.ts
import type { GeneratedProgram } from './programGenerator';

// Human-readable week, used by snapshot tests. Format is byte-identical to the Swift
// ProgramPrinter: note the em-dash "—" in the header and the trailing newline.
export function table(program: GeneratedProgram): string {
  const out: string[] = [];
  program.days.forEach((day, i) => {
    const total = day.lifts.reduce((s, l) => s + l.sets, 0);
    out.push(`Day ${i + 1} — ${day.lifts.length} lifts, ${total} sets`);
    for (const lift of day.lifts) {
      out.push(`  ${lift.exercise.name} [${lift.exercise.primary}] ${lift.sets} x ${lift.exercise.repRange}`);
    }
  });
  return out.join('\n') + '\n';
}
